tests/integration_tests/test_shell_plate_rotated.py

```diff
@@ -149,3 +149,32 @@ def elem_call_back(
             )
 
         return tacs_probs, fea_assembler
+
+    def test_residual_scaling(self):
+        """Test that the load scaling is working correctly for the point and pressure loads."""
+        for problem in self.tacs_probs:
+            with self.subTest(problem=problem.name):
+                np.random.seed(1)
+                # Check that the residual is zero if the states and load scale are both zero
+                problem.loadScale = 0.0
+                problem.u.zeroEntries()
+                problem.assembler.setVariables(problem.u)
+                problem.getResidual(problem.res)
```

Comment:
Could we simplify this down by adding `getJacobian`/`setLoadScale`/`addLoadScale` methods?

This portion of the test could look something like:
```
res = self.fea_assembler.createVec()
problem.setLoadScale(0.0)
problem.zeroVariables()
problem.getResidual(res)
```

Reply:
Good idea, I implemented `getJacobian` and `setLoadScale`, what would `addLoadScale` do?

Reply:
Sorry not sure where that last one came from, I meant `getLoadScale`

Reply:
Yes, lets make the `getJacobian` logic similar to whats in `getResidual`. Also could you return a copy of `self.K` rather than the actual pointer to `self.K`. I also think that by default it should return it as a scipy matrix, but could have an option to return a TACS matrix class

Reply:
I'm thinking something like this:

```
def getJacobian(self):
        """Get the problem's Jacobian in sciPy sparse matrix format
        Returns
        -------
        tuple of 2 or 4 scipy.sparse.bsr_matrices
            A tuple of 2 scipy.sparse.bsr_matrices (A, B) if Jacobian is a TACSParallelMat, or 4
            scipy.sparse.bsr_matrices (A, B, C, D) if Jacobian is a TACSSchurMat
        """
        # Make sure assembler variables are up-to-date
        self._updateAssemblerVars()
        # Make sure stiffness mat is up-to-date
        self._initializeSolve()
        # Return copy of scipy mat
        return copy.deepcopy(self.K.getMat())
```

Reply:
Sure, this is probably something for a later PR, but I was thinking it'd be good to split up the code that updates the stiffness matrix from the code that factors the PC. There are cases like this, and sometimes in a nonlinear solve where you just want to update the stiffness without doing the expensive refactorisation.

Reply:
No that definitely makes sense, let's include that split with the eventual PR for the nonlinear solver